### PR TITLE
Trim Reblogs: Mention legacy edits + legacy XKit in legacy post warning

### LINKED ISSUE #957: Trim Reblogs: Mention "New XKit" quick tags in legacy post warning
### Platform

MacOS 10.15.7 x86

### Browser

Ungoogled Chromium v109.0.5414.119

### Addon version

master

### Details

Reproducing my [Tumblr post](https://www.tumblr.com/transienturl/708568613642190848/oh-good-freaking-lord-using-legacy-new-xkit) below.

> oh good freaking lord.
> 
> using legacy "new xkit" quick tags on a post converts it to legacy (`is_blocks_post_format: false`), which breaks polls and presumably breaks xkit rewritten trim reblogs on that thread going forward.
> 
> if this is true and we don't want to basically hack xkit rewritten's quick tags backend into legacy "new xkit" or retire the damn thing's quick tags (which will upset people who like the old API), that will have to be added to the trim reblogs warning message and the great informative posts that've been written by people like [rpschtuff](https://rpschtuff.tumblr.com/), [shinsources](https://shinsources.tumblr.com/), and [rpclefairy](https://rpclefairy.tumblr.com/).
> 
> not sure what course of action is best here.

### PR DESCRIPTION
**Pull request by marcustyphoon**
### Description

This mentions an edit with the _legacy post editor or the_ previous XKit version as a blanket possible reason for a post appearing as legacy in the Trim Reblogs warning popup.

Resolves #957.

### Testing steps

Draft a reblog of  https://www.tumblr.com/transienttest/708567905092157440/beta-reblog-2. Try trimming it. Verify that the UI copy makes sense and looks good.

### CODE REVIEW

**Comment by AprilSylph** [comment]
- NPF reblog of NPF post = NPF
- Legacy reblog of NPF post = NPF
- NPF edit of NPF reblog of NPF post = NPF
- NPF edit of legacy reblog of NPF post = NPF
- Legacy edit of NPF reblog of NPF post = legacy
- Legacy edit of legacy reblog of NPF post = legacy

?

**Comment by marcustyphoon** [comment]
**NPF reblog of NPF post = NPF** 👍 
**Legacy reblog of NPF post = NPF** uh iirc basically
**NPF edit of NPF reblog of NPF post = NPF** 👍

**NPF edit of legacy reblog of NPF post = NPF**:

You are locked out of doing this unless you are on mobile. Doing that edit on mobile (erroneously?) sets shouldShowInLegacy to false, but leaves isBlocksPostFormat on false.

**Legacy edit of NPF reblog of NPF post = legacy**:

I don't think you can normally do this but I guess xkit 7 does so... yep?

**Legacy edit of legacy reblog of NPF post = legacy**:

actually wait now I'm confused. um.

**Comment by AprilSylph** [comment]
"NPF edit" can also mean using XKit Rewritten's Quick Tags, similarly "legacy edit" can also mean using New XKit's Quick Tags

**Comment by marcustyphoon** [comment]
Nevermind, I guess these are all allowed by the UI.

**NPF reblog of NPF post = NPF** confirmed
**Legacy reblog of NPF post = NPF** confirmed
**NPF edit of NPF reblog of NPF post = NPF** confirmed
**NPF edit of legacy reblog of NPF post = NPF** confirmed
**Legacy edit of NPF reblog of NPF post = legacy** confirmed
**Legacy edit of legacy reblog of NPF post = legacy** confirmed

File: src/scripts/trim_reblogs.js [hunk 1]
```diff
@@ -38,7 +38,9 @@ const onButtonClicked = async function ({ currentTarget: controlButton }) {
       showModal({
         title: 'Note: Legacy post',
         message: [
-          'The root post of this thread was originally created with the legacy post editor.',
+          'The root post of this thread was originally created with, or a post in it was edited with, the ',
```
Comment (AprilSylph) [inline src/scripts/trim_reblogs.js:41]:
```suggestion
          'This thread was originally created, or at some point edited, using the ',
```

File: src/scripts/trim_reblogs.js [hunk 2]
```diff
@@ -38,7 +38,9 @@ const onButtonClicked = async function ({ currentTarget: controlButton }) {
       showModal({
         title: 'Note: Legacy post',
         message: [
-          'The root post of this thread was originally created with the legacy post editor.',
+          'This thread was originally created, or at some point edited, using the ',
```
Comment (AprilSylph) [inline src/scripts/trim_reblogs.js:41]: lol you were too quick, I think this is more grammatically correct?
```suggestion
          'This thread was originally created, or at some point was edited, using the ',
```

Comment (marcustyphoon) [reply src/scripts/trim_reblogs.js:41]: wait how did github... what? it looked like that on my screen when I accepted the suggestion but it didn't do it :P

**Review by AprilSylph — APPROVED** [review]

**Pull request merged**